src/coreclr/nativeaot/System.Private.CoreLib/src/System/Activator.CoreRT.cs

```diff
@@ -29,47 +29,56 @@ public static partial class Activator
         // block so that the correct exceptions are generated. Also, it handles the cases where the T type doesn't have
         // a default constructor.
         [DebuggerGuidedStepThrough]
-        public static T CreateInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]T>()
+        public static unsafe T CreateInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]T>()
         {
-            if (!RuntimeHelpers.IsReference<T>())
+            if (RuntimeHelpers.IsValueTypeWithNoParameterlessConstructor<T>())
```

Comment:
Do I read this right that this path is just an optimization, but otherwise redundant with `ValueTypeWithNoConstructorMethod` marker method?

Comment may be useful. It is not obvious that `IsValueTypeWithNoParameterlessConstructor` can lie.

Reply:
Yeah, I put it there so that it keeps working with universal shared code.

Thinking about this, maybe I should put the struct without constructors on the same plan as CoreCLR (just rewrite the entire method body of `CreateInstance`). The added helper is not very general purpose.